File: C/Frame/Frame.c
#include<stdio.h>
#include<string.h>

void frame(char list[100][100], int length){
    int i,j,max=0,l,b[100];
    for(j=0;j<length;j++){
        l=strlen(list[j]);                    //Finding the length of each string and storing in an array
        b[j]=l;
        if(max<l)                             //Finding the length of the longest string in the list
            max=l;
    }
   
    //Printing in rectangle form
    for(j=0;j<max+4;j++)
        printf("*");
    printf("\n");

    for(i=0;i<length;i++){                   
        printf("*");
        printf(" %s ",list[i]);
        if(b[i]<max){
            for(j=0;j<(max-b[i]);j++)
                printf(" ");
        }
        printf("*");
        printf("\n");
    }
    

    for(j=0;j<max+4;j++)
        printf("*");
    printf("\n");
}

int main(){
    int len,i;
    char a[100][100];
    printf("Enter the length of the list: ");
    scanf("%d",&len);
    printf("\nEnter the strings:\n");
    for(i=0;i<len;i++){
        printf("%d: ",i+1);
        scanf("%s",a[i]);
    }
    frame(a,len);
}

File: Python/Frame/Frame.py
import sys
def frame(list):                                 #Function to print in a rectangular frame
    max=0 
    for x in list:                               #Finding length of the longest string in the list
        l=len(x)
        if(max<l):
            max=l                 

    j=0
    while(j<(max+4)):       
        sys.stdout.write("*")                    #Printing the top line of the rectangular frame
        j+=1
    sys.stdout.write("\n")

    for x in list:
        sys.stdout.write("*")
        sys.stdout.write(" "+x+" ")              #Printing the list elements
        if(len(x)<max):
            j=0
            while(j<(max-len(x))):
                sys.stdout.write(" ")
                j+=1
        sys.stdout.write("*")
        sys.stdout.write("\n")
    
    j=0
    while(j<(max+4)):
        sys.stdout.write("*")                    #Printing the bottom line of the rectangular frame
        j+=1
    sys.stdout.write("\n")  

length=input("Enter the length of the list:") 
i=0
L=[]
while(i<length):
    ele=raw_input("Enter element: ")             #Inputting the list elements
    L.append(ele)
    i+=1
frame(L)